model-optimizer/mo/front_ng/extractor.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging as log
+import re
+from collections import defaultdict
+from copy import copy
+
+import numpy as np
+
+from mo.utils.error import Error
+
+from ngraph.frontend import InputModel  # pylint: disable=no-name-in-module,import-error
+
+
+def decode_name_with_port(input_model: InputModel, node_name: str):
+    """
+    Decode name with optional port specification w/o traversing all the nodes in the graph
+    TODO: in future node_name can specify input/output port groups and indexes
```

Comment:
nit: `indexes` -> `indices`. The same comment for the line 30

Reply:
Fixed